akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -224,7 +254,18 @@ object Framing {
     val lengthFieldLength:  Int,
     val lengthFieldOffset:  Int,
     val maximumFrameLength: Int,
-    val byteOrder:          ByteOrder) extends GraphStage[FlowShape[ByteString, ByteString]] {
+    val byteOrder:          ByteOrder,
+    computeFrameSize:       Option[(Array[Byte], Int) ⇒ Int]) extends GraphStage[FlowShape[ByteString, ByteString]] {
```

Comment:
I think, pattern match is better, it makes code more readable and you don't need to keep cache.